ts/features/bonus/siciliaVola/saga/networking/handleGetAereoportiBeneficiario.ts

```diff
@@ -0,0 +1,69 @@
+import { ActionType } from "typesafe-actions";
+import { call, put } from "redux-saga/effects";
+import { BackendSiciliaVolaClient } from "../../api/backendSiciliaVola";
+import { getGenericError, getNetworkError } from "../../../../../utils/errors";
+import { SagaCallReturnType } from "../../../../../types/utils";
+import { isDefined } from "../../../../../utils/guards";
+import { SessionManager } from "../../../../../utils/SessionManager";
+import { MitVoucherToken } from "../../../../../../definitions/io_sicilia_vola_token/MitVoucherToken";
+import { svGenerateVoucherAvailableDestination } from "../../store/actions/voucherGeneration";
+import { AvailableDestinations } from "../../types/SvVoucherRequest";
+import { AeroportoSedeBeanList } from "../../../../../../definitions/api_sicilia_vola/AeroportoSedeBeanList";
+
+// TODO: add the mapping when the swagger will be fixed
+const mapKinds: Record<number, string> = {};
+
+// convert a success response to the logical app representation of it
+const convertSuccess = (
+  availableDestinations: AeroportoSedeBeanList
+): AvailableDestinations =>
+  availableDestinations
+    .map(d => (d.denominazione ? d.denominazione : undefined))
+    .filter(isDefined);
+
+export function* handleGetAereoportiBeneficiario(
+  getAeroportiBeneficiario:
+    | ReturnType<typeof BackendSiciliaVolaClient>["getAeroportiBeneficiario"]
+    | ReturnType<typeof BackendSiciliaVolaClient>["getAeroportiStato"],
+  svSessionManager: SessionManager<MitVoucherToken>,
+  action: ActionType<typeof svGenerateVoucherAvailableDestination.request>
+) {
+  try {
+    const request = svSessionManager.withRefresh(
+      getAeroportiBeneficiario(action.payload)
+    );
+    const getAeroportiBeneficiarioResult: SagaCallReturnType<typeof request> = yield call(
+      request
+    );
+
+    if (getAeroportiBeneficiarioResult.isRight()) {
+      if (getAeroportiBeneficiarioResult.value.status === 200) {
+        yield put(
+          svGenerateVoucherAvailableDestination.success(
+            convertSuccess(getAeroportiBeneficiarioResult.value.value)
+          )
+        );
+      }
+      if (mapKinds[getAeroportiBeneficiarioResult.value.status] !== undefined) {
+        yield put(
+          svGenerateVoucherAvailableDestination.failure({
+            ...getGenericError(
+              new Error(mapKinds[getAeroportiBeneficiarioResult.value.status])
+            )
+          })
+        );
+      }
+    } else {
+      // cannot decode response
+      yield put(
+        svGenerateVoucherAvailableDestination.failure({
+          ...getGenericError(new Error("Generic Error"))
```

Comment:
Not sure how error tracking works at this level, but I see more useful a more specific message like `new Error("invalid payload from getAeroportiBeneficiarioResult")`.
It will stand out in logs and reports.

Reply:
Yes, I think it's better. Updated in [39ae42c](https://github.com/pagopa/io-app/pull/3267/commits/39ae42ced9898bdeb52cc6fe768b199249636363)